### ISSUE TITLE
Document different versions of GridFunction error computation functions

### ISSUE THREAD

**Issue by dylan-copeland** [mfem/mfem#4057]
As discussed in https://github.com/mfem/mfem/pull/3746, it is undocumented and unclear to me as to why different functions are called by `GridFunction::ComputeL1Error` depending on whether the input `Coefficient` is passed by reference or as a pointer. Some questions are:

1. Why use different functions, namely `ComputeW11Error` and `ComputeLpError`?
2. Are the computations the same up to round-off error?
3. Are the different versions intended to be used differently in practice?

The same goes for `ComputeL2Error`. If someone is familiar with this, it would be helpful to document these functions.

**Comment by v-dobrev** [comment]
@dylan-copeland, @mlstowell, do you think it makes sense to address the questions and concerns from this issue in #4578?

**Comment by dylan-copeland** [comment]
@v-dobrev @mlstowell I agree, it would be good to resolve this issue in that PR. Thanks for the reminder!

**Comment by mlstowell** [comment]
It's not clear to me why these two versions of `ComputeL1Error` call different functions internally. 

I do _not_ expect the results to agree to round-off error because they use different order integration rules; `ComputeW11Error` uses `2p+1` whereas `ComputeLpError` uses `2p+3`. There are other implementation differences but these probably only change the order of operations (so these probably are round-off).

I have no idea about intended use differences. I suspect most developers don't realize they are different at all.

Regarding PR #4578: I would be happy to expand the documentation on these and related member functions in this PR. I can also alter one or both versions of `ComputeL1Error` but I'm not sure which implementation to use. @v-dobrev, what do you think is best here?
